python/ray/tune/execution/tune_controller.py

```diff
@@ -549,6 +549,12 @@ def _maybe_add_actors(candidates: List[Trial]):
                 )
                 if not self._maybe_reuse_cached_actor(start_trial):
                     self._resources_to_pending_trials[resource].add(start_trial)
+                else:
+                    if start_trial not in self._staged_trials:
+                        self._staged_trials.add(start_trial)
+                        self._actor_cache.increase_max(
+                            start_trial.placement_group_factory
+                        )
```

Comment:
Why do we need to increase the actor cache max here? It reused the actor, so there's no need to cache another actor for this trial.

Reply:
A1 trial staged, A2 trial staged --> max cached actors = 2
A1 actor started --> unstaged --> max cached actors = 1
A1 actor stopped --> cached for reuse since A2 is staged
A2 actor started --> unstaged --> max cached actors = 0
A2 actor stopped --> not cached for reuse
A3 is pending and can be started with A1's cached actor
   - If we don't add A3 to staged trials, it'll encounter an error when we unstage the trial in `on_trial_reset` --> `actor_started` --> `unstage_trial_with_resources` (that's why we do `self._staged_trials.add(start_trial)`)
   - If we don't increase the max actors in the cache, it'll encounter an error when we unstage the trial and decrease the max cached actors to -1. That's why we add `self._actor_cache.increase_max(...)`.